### PR TITLE
chore: Bring back me.saleRegistrations as a connection

### PR DESCRIPTION
**Pull request by jo-rs**
Part 1 of 2 for resolving [PURCHASE-2185](https://artsyproduct.atlassian.net/browse/PURCHASE-2185).

Commenting `me.saleRegistrations` field back in to allow for move from `v1` to `v2` on `force`, and turning it into a connection.

Follow up on [this](https://github.com/artsy/force/pull/6810) PR.

### CODE REVIEW

**Comment by sepans** [comment]
I did a little bit of digging and found [this comment](https://github.com/artsy/metaphysics/pull/1927/commits/6f4ed76a2259e1f60b34efcf75e5a177eb9199c8?file-filters%5B%5D=.ts#diff-d6c389cc2c364669873566158557d24d63d25d5ea31e4846614d106a37a94f05R31). So I _think_ @alloy commented this out so we change this to a connection instead of a list when we start to use it again. Sorry this ticket ended up being more involved than we originally thought 😝

**Comment by jo-rs** [comment]
> I did a little bit of digging and found [this comment](https://github.com/artsy/metaphysics/pull/1927/commits/6f4ed76a2259e1f60b34efcf75e5a177eb9199c8?file-filters%5B%5D=.ts#diff-d6c389cc2c364669873566158557d24d63d25d5ea31e4846614d106a37a94f05R31). So I _think_ @alloy commented this out so we change this to a connection instead of a list when we start to use it again. Sorry this ticket ended up being more involved than we originally thought 😝

Oh, I see. Ok, switching to WIP.

**Comment by zephraph** [comment]
Great! Thanks for this @jo-rs and sorry for the delay on the review.

File: _schemaV2.graphql [hunk 1]
```diff
@@ -9652,6 +9677,36 @@ type SaleEdge {
...
+  sale: Sale
+}
+
+# A connection to a list of items.
+type SaleRegistrationConnection {
+  # A list of edges.
+  edges: [SaleRegistrationEdge]
+
+  # Information to aid in pagination.
+  pageInfo: PageInfo!
```
Comment (sepans) [inline _schemaV2.graphql:9698]: Sorry I was slow responding to this PR. Is there a reason we have `pageInfo` here and not `pageCursor`?

Comment (zephraph) [reply _schemaV2.graphql:9698]: `pageInfo` is the correct nomenclature. Here's the [relay pagination spec](https://relay.dev/graphql/connections.htm#sel-GABDJAaBHBPwxL).

**Pull request merged**